src/zap/zapinfo.cpp

```diff
@@ -2083,6 +2102,59 @@ void ZapInfo::GetProfilingHandle(BOOL                      *pbHookFunction,
     *pbIndirectedHandles = TRUE;
 }
 
+DWORD FilterMethodAttribsForIsSupported(DWORD attribs, CORINFO_METHOD_HANDLE ftn, ICorDynamicInfo* pJitInfo)
+{
+    if (attribs & CORINFO_FLG_JIT_INTRINSIC)
+    {
+        // Do not report the get_IsSupported method as an intrinsic. This will turn the call into a regular
+        // call. We also make sure none of the hardware intrinsic method bodies get pregenerated in crossgen
+        // (see ZapInfo::CompileMethod) but get JITted instead. The JITted method will have the correct
+        // answer for the CPU the code is running on.
+
+        const char* namespaceName;
+        const char* className;
+        const char* enclosingClassName;
+        const char* methodName = pJitInfo->getMethodNameFromMetadata(ftn, &className, &namespaceName, &enclosingClassName);
+
+        // If it's not the IsSupported method, we're done
+        if (strcmp(methodName, "get_IsSupported") != 0)
+            return attribs;
+
+        bool isX86intrinsic = strcmp(namespaceName, "System.Runtime.Intrinsics.X86") == 0;
+
+#if defined(_TARGET_X86_) || defined(_TARGET_AMD64_)
+        // If it's IsSupported on Sse/Sse2, we can expand unconditionally since this is reliably
+        // available everywhere.
+        if (isX86intrinsic
```

Comment:
Just to confirm, this means that for R2R code, SSE/SSE2 will be unconditionally used (on x86/x64) and users "should" still be able to control higher ISAs (such as `SSE3`) via the `COMPlus_Enable<Isa>` knobs.

For code that is rejitted, all `COMPlus_Enable<Isa>` knobs should be supported, including SSE/SSE2.

Is that correct?

Reply:
Yes, that should be it.